### PR TITLE
Move mutation_rate into EA

### LINKED ISSUE #256: `mutation_rate` should be a parameter of the EA not the population
Currently `mutation_rate` is a parameter for the constructor of a `Population` object. However, it would be more consistent to associate this with the evolutionary algorithms objects. Concretely, it should be moved to `MuPlusLambda`.

### PR DESCRIPTION
**Pull request by HenrikMettler**
Moves the parameter mutation_rate from population to mu_plus_lambda. Closes #256

### CODE REVIEW

File: test/test_population.py [hunk 1]
```diff
@@ -29,12 +18,12 @@ def test_champion(population_simple_fitness):
 
 
 def test_mutate(population_params, genome_params):
```
Comment (jakobj) [inline test/test_population.py:20]: if you move `mutate` this test should also be moved to the mupluslambda tests

**Review by jakobj — CHANGES_REQUESTED** [review]
awesome, thanks for working on this!

a few inline comments that should be addressed.

File: test/conftest.py [hunk 2]
```diff
@@ -59,3 +63,13 @@ def population_simple_fitness(population_params, genome_params):
 @fixture
 def local_search_params():
     return {"lr": 1e-3, "gradient_steps": 9}
+
+
+@fixture
+def n_offsprings():
```
[the remark this answers is not in the document]
Comment (HenrikMettler) [reply test/conftest.py:69]: I use them sperately in `test_assert_mutation_rate` (not as `ea_params`) (can also remove it, but i find it confusing if there are numerical values in the test)

File: cgp/ea/mu_plus_lambda.py [hunk 3]
```diff
@@ -242,3 +250,23 @@ def update_n_objective_calls(self, combined: List[IndividualBase]) -> None:
...
             if individual.fitness_is_None():
                 self.n_objective_calls += 1
+
+    def mutate(
+        self, offsprings: List[IndividualBase], rng: np.random.RandomState
+    ) -> List[IndividualBase]:
+        """Mutate a list of offspring individuals.
+
+        Parameters
+        ----------rng
```
Comment (jakobj) [inline cgp/ea/mu_plus_lambda.py:260]:
```suggestion
        ----------
```

**Review by jakobj — APPROVED** [review]
great stuff :+1:

one type, then please squash into one commit before merging

**Comment by jakobj** [comment]
unfortunately there are now some conflicts. please rebase and fix conflicts

File: cgp/ea/mu_plus_lambda.py [hunk 4]
```diff
@@ -34,8 +35,15 @@ def __init__(
         ----------
         n_offsprings : int
             Number of offspring in each iteration.
+<<<<<<< HEAD
```
Comment (jakobj) [inline cgp/ea/mu_plus_lambda.py:38]: hmm, something went wrong here

Comment (HenrikMettler) [reply cgp/ea/mu_plus_lambda.py]: oops didn't check the docstring while merging

**Comment by jakobj** [comment]
excellent :+1: and merging

**Pull request merged**